task-5.1-code-update: added table component with filtering asc/desc

vanshika-dargan-18 opened the pull request:
Task Overview:

Created a Reusuable Table Component Similar to One in VBC Admin Portal. It also implements sorting in ascending or descending order. Populated with sample data as of now coming from json/tableData.json. 
<img width="1512" alt="Screenshot 2024-12-24 at 3 05 57 PM" src="https://github.com/user-attachments/assets/8e9f45e7-4e1c-4bc7-b136-576fa7d35f77" />

pratikjoshivonage commented on src/components/dashboard/body/AdminBody.vue:
```diff
@@ -23,6 +24,12 @@ export default {
     TableComponent,
   },
+  data() {
+    return {
```
please simplify with spread operator
```
return {
  ...tableData,
};

```

pratikjoshivonage commented on src/components/common/TableComponent.vue:
```diff
@@ -1,11 +1,128 @@
+.table-container th:before {
+  content: '';
+  display: inline-block;
+  background-image: url('../../assets/both-arrows.svg');
```
please use absolute url instead of relative.

pratikjoshivonage commented on src/components/common/TableComponent.vue:
```diff
@@ -1,11 +1,128 @@
+}
+
+.table-container th[data-sort='asc']:before {
+  background-image: url('../../assets/up-arrow.svg');
```
please use absolute url instead of relative.

pratikjoshivonage commented on src/components/common/TableComponent.vue:
```diff
@@ -1,11 +1,128 @@
+    };
+  },
+  computed: {
+    sortedData() {
```
We can update the function.

    sortedData() {
      if (!this.sortKey) return this.data;
      return [...this.data].sort((a, b) => {
        const comparison = a[this.sortKey] > b[this.sortKey] ? 1 : -1;
        return this.sortOrder === 'asc' ? comparison : -comparison;
      });
    }

vanshika-dargan-18 commented:
@pratikjoshivonage I have addressed all issues, please check once.

pratikjoshivonage commented:
> @pratikjoshivonage I have addressed all issues, please check once.

Can you please check conflicts, fetch and rebase from main branch to take https://github.com/Admin-Portal-VDS/Admin-Portal/pull/19#event-15763762477 files from this commit.

vanshika-dargan-18 commented:
> > @pratikjoshivonage I have addressed all issues, please check once.
> 
> Can you please check conflicts, fetch and rebase from main branch to take [#19 (comment)](https://github.com/Admin-Portal-VDS/Admin-Portal/pull/19#event-15763762477) files from this commit.

Yes doing that..
